recipes/libyang/all/conanfile.py

```diff
@@ -0,0 +1,79 @@
+import os
+
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout, CMakeDeps
+from conan.tools.files import get, rmdir, copy
+
+required_conan_version = ">=1.53.0"
+
+
+class LibYangConan(ConanFile):
+    name = "libyang"
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "YANG data modeling language library"
+    homepage = "https://github.com/CESNET/libyang"
+    topics = ("yang", "bsd", "netconf", "restconf", "yin")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False],
+               "fPIC": [True, False]}
+    default_options = {
+        "shared": False,
+        "fPIC": True
+    }
+
+    tool_requires = "cmake/[>=3.22.0 <4]"
+
+    def requirements(self):
+        self.requires("pcre2/10.42", transitive_headers=True)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["ENABLE_TESTS"] = False
+        tc.variables["ENABLE_VALGRIND_TESTS"] = False
+        tc.variables["ENABLE_STATIC"] = not self.options.shared
```

Comment:
```suggestion
        tc.variables["ENABLE_STATIC"] = not self.options.shared
        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
```
Needed for correct functioning of `tc.variables` with `cmake_minimum_required(VERSION 2.8.12)`.